src/client-options-converter.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  EppoClientParameters,
+  EventDispatcher,
+  Flag,
+  IConfigurationStore,
+} from '@eppo/js-client-sdk-common';
+
+import { IClientOptions } from './i-client-config';
+import { sdkName, sdkVersion } from './sdk-data';
+
+/**
+ * Converts IClientOptions to EppoClientParameters
+ * @internal
+ */
+export function clientOptionsToEppoClientParameters(
+  options: IClientOptions,
+  flagConfigurationStore: IConfigurationStore<Flag>,
+  eventDispatcher?: EventDispatcher,
+): EppoClientParameters {
+  const parameters: EppoClientParameters = {
+    flagConfigurationStore,
+    isObfuscated: true,
+  };
+
+  parameters.eventDispatcher = eventDispatcher;
+
+  // Always include configuration request parameters
+  parameters.configurationRequestParameters = {
+    apiKey: options.sdkKey,
+    sdkVersion, // dynamically picks up version.
+    sdkName, // Hardcoded to `js-client-sdk`
+    baseUrl: options.baseUrl,
+    requestTimeoutMs: options.requestTimeoutMs,
+    numInitialRequestRetries: options.numInitialRequestRetries,
+    numPollRequestRetries: options.numPollRequestRetries,
+    pollAfterSuccessfulInitialization: options.pollAfterSuccessfulInitialization,
+    pollAfterFailedInitialization: options.pollAfterFailedInitialization,
+    pollingIntervalMs: options.pollingIntervalMs,
+    throwOnFailedInitialization: options.throwOnFailedInitialization,
+    skipInitialPoll: options.skipInitialRequest,
+  };
```

Comment:
This seems like it could e a pain to keep in sync. If these parameters are just a supserset, could we use the spread operator instead?
```ts
const parameters = {...options, eventDispatcher, isObfuscated: true}
```

Side note: I've heard a couple of places they'd like to be able to tell client _not_ to obfuscate, such as in development environments. So I wonder if we take this opportunity to make `isObfuscated` configurable.

Reply:
removed

Reply:
https://github.com/Eppo-exp/js-sdk-common/pull/218